spec/features/users/dashboard_spec.rb

```diff
@@ -225,6 +208,35 @@
 
         expect(page).to have_current_path("/gyms?zip_code=#{@user.zip_code}")
       end
+
+      it 'displays my upcoming workouts', :vcr do
+        expect(page).to have_css('#upcoming-workouts')
+
+        within '#upcoming-workouts' do
+          expect(page).to have_content('Upcoming Workouts')
+
+          user_events.each do |event|
+            expect(page).to have_css("#event-#{event.id}")
+
+            within "#event-#{event.id}" do
+              expect(page).to have_content(event.activity)
+              expect(page).to have_link('Delete')
+            end
+          end
+        end
+      end
+
+      it 'displays a link to delete a workout', :vcr do
+        allow(BackEndService).to receive(:delete_event).and_return(204)
```

Comment:
Exactly what I was gonna do when I stub out a patch method. So I believe this is fine.